model_analyzer/analyzer.py

```diff
@@ -213,6 +217,12 @@ def _create_model_manager(self, client, gpus):
             state_manager=self._state_manager)
 
     def _get_server_only_metrics(self, client):
+        if self._do_checkpoint_server_metrics_match():
```

Comment:
bare minimum this should be inside of the `!= c_api` code below. However, I think this will also not work for remote mode.  Then again, remote mode server will already be up and we might be able to solve that case slightly differently?

Reply:
I've tried this in remote mode and it does work as we are able to acquire the gpuids